problems/arraysWithLoops.js

```diff
@@ -25,23 +44,44 @@ function shoutWhileLoop() {}
  * @returns {Array} - ["I!", "am!", "a!", "happy!", "array!"]
  */
 
-function shoutForOfLoop() {}
+const shoutForOfLoop = (array) => {
+  let newArray = []
+  for(element of array) {
+    newArray.push(element + "!")
+  }
+return newArray
+}
 
 /**
  * Takes in an array and returns the sum of all values
  * @param {number[]} nums
  * @returns {number} sum
  */
 
-function sumArray() {}
+const sumArray = (nums) => {
+  let sum = 0
+  for(let i = 0; i < nums.length; i++) {
+    sum += nums[i]
+  }
+  return sum
+}
 
 /**
  * Takes in an array and returns an array of all the odd valued elements
  * @param {number[]} nums
  * @returns {array} odds
  */
 
-function oddArray() {}
+const oddArray = (nums) => {
+  let newArray = nums.slice(0)
```

Comment:
Don't slice here. That will create a copy of all elements are with empty array and populate with odd.